mlrun/api/api/endpoints/functions.py

```diff
@@ -885,3 +894,26 @@ def _process_model_monitoring_secret(db_session, project_name: str, secret_key:
         Secrets().delete_project_secret(project_name, provider, secret_key)
 
     return secret_value
+
+
+def _is_nuclio_deploy_status_changed(
+    previous_status: dict, new_status: dict, new_state: str, new_nuclio_name: str = None
+) -> bool:
+    # get relevant fields from the new status
+    new_container_image = new_status.get("containerImage", "")
+    new_internal_invocation_urls = new_status.get("internalInvocationUrls", [])
+    new_external_invocation_urls = new_status.get("externalInvocationUrls", [])
+    address = new_external_invocation_urls[0] if new_external_invocation_urls else ""
+
+    # Determine if any of the relevant fields have changed
+    has_changed = (
+        previous_status.get("functionName", "") != new_nuclio_name
+        or previous_status.get("state") != new_state
+        or previous_status.get("containerImage", "") != new_container_image
+        or previous_status.get("internalInvocationUrls", [])
+        != new_internal_invocation_urls
+        or previous_status.get("externalInvocationUrls", [])
+        != new_external_invocation_urls
+        or previous_status.get("address", "") != address
+    )
```

Comment:
why not take status block, dump it to json string, md5 it, store as "hash"
on next iteration, get nuclio function, do it again and compare hashes.

Reply:
talked offline - my approach won't work because mlrun keeps a resolved status (read: mlrun function status) that have different structure anyway, thus, it wont work).